test/e2e/minscale_readiness_test.go

```diff
@@ -138,15 +146,19 @@ func latestRevisionName(t *testing.T, clients *test.Clients, configName string)
 	return config.Status.LatestCreatedRevisionName
 }
 
-func waitForDesiredScale(t *testing.T, clients *test.Clients, deploymentName string, cond func(int32) bool) error {
+func waitForDesiredScale(t *testing.T, clients *test.Clients, deploymentName string, privateServiceName string, cond func(int32) bool) error {
 	deployments := clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace)
+	endpoints := clients.KubeClient.Kube.CoreV1().Endpoints(test.ServingNamespace)
 
 	return wait.PollImmediate(time.Second, 1*time.Minute, func() (bool, error) {
 		deployment, err := deployments.Get(deploymentName, metav1.GetOptions{})
```

Comment:
I don't think fetching the deployment is actually required anymore. Shall we remove all mentions of it and redo the error messages above to not include it as well?

Reply:
Yes. If endpoints satisfy the scale, the deploy satisfies too.